src/main/java/com/epam/aidial/core/upstream/LoadBalancerProvider.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.aidial.core.upstream;
+
+import com.epam.aidial.core.config.Upstream;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class LoadBalancerProvider {
+
+    private final Cache<String, TieredBalancer> cache = CacheBuilder.newBuilder()
+            .expireAfterAccess(5, TimeUnit.MINUTES)
```

Comment:
We shouldn’t evict entries after 5 mins: consider back off is greater than 5 mins